xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -318,12 +318,20 @@ func (b *cdsBalancer) handleWatchUpdate(update clusterHandlerUpdate) {
 				EDSServiceName:        cu.EDSServiceName,
 				MaxConcurrentRequests: cu.MaxRequests,
 			}
-			if cu.EnableLRS {
-				// An empty string here indicates that the cluster_resolver balancer should use the
-				// same xDS server for load reporting as it does for EDS
-				// requests/responses.
-				dms[i].LoadReportingServerName = new(string)
-
+			if cu.LRSServerConfig == xdsresource.ClusterLRSServerSelf {
+				bootstrapConfig := b.xdsClient.BootstrapConfig()
+				parsedName := xdsresource.ParseName(cu.ClusterName)
+				if parsedName.Scheme == xdsresource.FederationScheme {
+					// Is a federation resource name, find the corresponding
+					// authority server config.
+					if cfg, ok := bootstrapConfig.Authorities[parsedName.Authority]; ok {
+						dms[i].LoadReportingServer = cfg.XDSServer
```

Comment:
The gFRC has the following line:
```
Note that we currently do not want to trust the control plane to tell us to contact any arbitrary server with
GoogleDefaultCredentials, since this could leak the oauth token. We also don't want that attack to happen via a
service config distributed via a DNS TXT record that configures the xds_cluster_impl LB policy to send load reports
to some arbitrary server with GoogleDefaultCredentials. To avoid this, the XdsClient load reporting APIs will require
that the LRS server passed into them is one of the servers that is already configured in the bootstrap file (i.e., it 
matches server name, channel creds, and server features).
```

Where would we do this check? Is this check even required now?